stages/org.osbuild.chmod

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+"""
+Change file mode bits
+
+Change the file mode bits of one or more files or directories inside the tree.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+from osbuild.util.path import in_tree
+
+SCHEMA = r"""
+"items": {
+  "type": "object",
+  "additionalProperties": false,
+  "patternProperties": {
+    "^\\/(?!\\.\\.)((?!\\/\\.\\.\\/).)+$": {
+      "type": "object",
+      "required": ["mode"],
+      "properties": {
+        "mode": {
+          "type": "string",
+          "description": "Symbolic or numeric octal mode"
+        },
+        "recursive": {
+          "type": "boolean",
+          "description": "Change modes recursively",
+          "default": false
+        }
+      }
+    }
+  }
+}
+"""
+
+def chmod(path: str, mode: str, recursive: bool) -> int:
+    arguments = [mode]
+    if recursive:
+        arguments.append("--recursive")
+    arguments.append("--")
+    arguments.append(path)
+
+    return subprocess.run(["chmod", *arguments], check=False).returncode
+
+def main(tree, options):
+    for path, cmdargs in options["items"].items():
+        mode = cmdargs["mode"]
+        recursive = cmdargs.get("recursive", False)
+        realpath = os.path.join(tree, path.lstrip("/"))
+        if not in_tree(realpath, tree, must_exist=True):
+            print(f"Error: path {path} not in tree")
+            return 1
```

Comment:
I think I would prefer an exception here. In the future we can catch those via and send them to osbuild in a structured manner. `print` means it lands in in the log and one has to scroll and look for it.